### PR TITLE
created look up in map function

### PR DESCRIPTION
**Pull request by gchq-21716**

### CODE REVIEW

File: core/src/main/java/uk/gov/gchq/koryphe/impl/function/LookUpInMap.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+import java.util.Map;
+
+/**
+ * A {@link LookUpInMap} is a {@link KorypheFunction} that takes a key and using a parameterised map returns a value.
+ */
+@Since("1.6.1")
```
Comment (d47853) [inline core/src/main/java/uk/gov/gchq/koryphe/impl/function/LookUpInMap.java]:
```suggestion
@Since("1.7.0")
```

File: core/src/main/java/uk/gov/gchq/koryphe/impl/function/LookUpInMap.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+@Summary("Looks up a value in a parameterised map")
```
Comment (d47853) [inline core/src/main/java/uk/gov/gchq/koryphe/impl/function/LookUpInMap.java]:
```suggestion
@Summary("Looks up a value in a map")
```

File: core/src/main/java/uk/gov/gchq/koryphe/impl/function/LookUpInMap.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+public class LookUpInMap<K, V> extends KorypheFunction<K, V> {
```
Comment (d47853) [inline core/src/main/java/uk/gov/gchq/koryphe/impl/function/LookUpInMap.java]: Can we change the name of the function to something like DictionaryLookup?

File: core/src/test/java/uk/gov/gchq/koryphe/impl/function/LookUpInMapTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Before
+    public void setup() {
+        lookUpMap.put("one", 1);
+        lookUpMap.put("two", 2);
+        lookUpInMap = new LookUpInMap<>(lookUpMap);
+    }
+
+    @Test
+    public void apply() {
```
Comment (d47853) [inline core/src/test/java/uk/gov/gchq/koryphe/impl/function/LookUpInMapTest.java]:
```suggestion
    public void shouldReturnExistingValueInTheMap() {
```

File: core/src/test/java/uk/gov/gchq/koryphe/impl/function/LookUpInMapTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        assertEquals(1, (int)lookUpInMap.apply("one"));
+        assertEquals(2, (int)lookUpInMap.apply("two"));
+    }
+
```
Comment (d47853) [inline core/src/test/java/uk/gov/gchq/koryphe/impl/function/LookUpInMapTest.java]: Can we add a test for when the Function is applied to a null value, null map, and value which doesn't exist in the map

File: core/src/main/java/uk/gov/gchq/koryphe/impl/function/DictionaryLookUp.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+import java.util.Map;
+
+/**
+ * A {@link DictionaryLookUp} is a {@link KorypheFunction} that takes a key and using a dictionary returns a value.
+ */
+@Since("1.7.0")
+@Summary("Looks up a value in a map")
+public class DictionaryLookUp<K, V> extends KorypheFunction<K, V> {
```
Comment (d47853) [inline core/src/main/java/uk/gov/gchq/koryphe/impl/function/DictionaryLookUp.java]: LookUp should be Lookup (without capital U)

File: core/src/test/java/uk/gov/gchq/koryphe/impl/function/DictionaryLookUpTest.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    public void lookUpNotFoundInDictionary() {
+        assertNull(dictionaryLookUp.apply("three"));
+    }
+
+    @Test
+    public void lookUpWithoutDictionary() {
+        try {
+            new DictionaryLookUp<>().apply("four");
+            Assert.fail("expected NullPointerException");
+        } catch (NullPointerException e) {}
```
Comment (d47853) [inline core/src/test/java/uk/gov/gchq/koryphe/impl/function/DictionaryLookUpTest.java]: Should throw an IllegalArgumentException here with useful error message

File: core/src/test/java/uk/gov/gchq/koryphe/impl/function/DictionaryLookUpTest.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        assertEquals(2, (int) dictionaryLookUp.apply("two"));
+    }
+
+    @Test
+    public void lookUpNullInDictionary() {
+        assertNull(dictionaryLookUp.apply(null));
+    }
+
+    @Test
+    public void lookUpNotFoundInDictionary() {
```
Comment (d47853) [inline core/src/test/java/uk/gov/gchq/koryphe/impl/function/DictionaryLookUpTest.java]:
```suggestion
    public void shouldReturnNullIfItemDoesntExistInDictionary() {
```

File: core/src/test/java/uk/gov/gchq/koryphe/impl/function/DictionaryLookUpTest.java [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        assertNull(dictionaryLookUp.apply("three"));
+    }
+
+    @Test
+    public void lookUpWithoutDictionary() {
```
Comment (d47853) [inline core/src/test/java/uk/gov/gchq/koryphe/impl/function/DictionaryLookUpTest.java]:
```suggestion
    public void shouldThrowExceptionIfDictionaryIsSetToNull() {
```
